prompttools/experiment/experiments/qdrant_experiment.py

```diff
@@ -0,0 +1,181 @@
+                raise RuntimeError(f"'{arg_name}' must be a frozen parameter in QdrantExperiment.")
+        return cls(**test_parameters, **frozen_parameters)
+
+    def qdrant_completion_fn(self, **kwargs) -> Dict[str, Any]:
+        query_result = self.client.search(self.collection_name, **kwargs)
+        return query_result
+
+    def prepare(self) -> None:
+        from qdrant_client import models
+
+        # Vectorize the documents and queries
+        self.vectorized_documents = [
+            models.Record(
+                id=i,
+                vector=self.embedding_fn(document),
+            )
+            for i, document in enumerate(self.documents)
+        ]
+        self.query_params["query_vector"] = [self.embedding_fn(query) for query in self.queries]
+
+        self.collection_args_combo: list[dict] = []
+        for combo in itertools.product(*self.collection_params.values()):
+            self.collection_args_combo.append(dict(zip(self.collection_params.keys(), combo)))
+
+        self.query_argument_combos: list[dict] = []
+        for combo in itertools.product(*self.query_params.values()):
+            self.query_argument_combos.append(dict(zip(self.query_params.keys(), combo)))
+
+    def run(self, runs: int = 1) -> None:
+        from qdrant_client import models
+
+        self.results = []
+        if not self.query_argument_combos:
+            logging.info("Preparing first...")
+            self.prepare()
+        for collection_args in self.collection_args_combo:
+            collection_args = self._create_nested_object(collection_args)
+
+            try:
+                # Create the collection
+                self.client.create_collection(self.collection_name, **collection_args)
+
+                # Upload and index the documents
+                self.client.upload_records(self.collection_name, self.vectorized_documents)
+
+                # Wait for the collection to be indexed
+                while True:
+                    collection_info = self.client.get_collection(self.collection_name)
+                    if collection_info.status == models.CollectionStatus.GREEN:
+                        logging.info("Collection is indexed")
+                        break
+                    logging.info("Waiting for collection to be indexed...")
+                    time.sleep(1)
+
+                # Run the queries
+                for query_args in self.query_argument_combos:
+                    query_args = self._create_nested_object(query_args)
+                    for _ in range(runs):
+                        self.queue.enqueue(self.completion_fn, query_args)
+
+                self.results.extend(self.queue.results())
+                self.scores["latency"].extend(self.queue.latencies())
+                if len(self.results) == 0:
+                    logging.error("No results. Something went wrong.")
+                    raise PromptExperimentException
```

Comment:
To save the results, you want to call `_construct_result_dfs(...)` here. 

See this updated [source code](https://github.com/hegelai/prompttools/blob/main/prompttools/experiment/experiments/experiment.py#L178).